app/models/assignment.rb

```diff
@@ -70,4 +70,20 @@ def to_label
     label += " - #{period_start && I18n.l(period_start)}"
     label
   end
+
+  def hours_since_last_submitted
+    if submitted_at.present?
+      hours.where('created_at > ?', submitted_at)
+    else
+      hours
+    end
+  end
+
+  def feedbacks_since_last_submitted
+    if submitted_at.present? && volunteer.user
+      feedbacks.where('created_at > ? AND author_id = ?', submitted_at, volunteer.user.id)
```

Comment:
And the Feedbacks scope can check for `submitted_at` in its relation for itself, because it can reach the needed information:

```ruby
# Feedback
scope :some_name, lambda {
  if feedbackable.submitted_at.present?
    where('created_at > ?', feedbackable.submitted_at).where(author: volunteer.user)
  else
    where(author: volunteer.user)
  end
}

# AssignmentController
@assignment.feedbacks.some_name
```

Reply:
Actually that was just bullshit... Sequel will not know what feedbackable is... 🤦‍♂️ 

but anyhow:

Solving what is possible within a scope inside `Feedback` would be the way to go.
In this case it could also be somewhat in that manner:

```ruby
scope :some_name, lambda { |submitted_at, volunteer_user|
  if submitted_at.present?
    where('created_at > ?', feedbackable.submitted_at).where(author: volunteer_user)
  else
    where(author: volunteer_user)
  end
}
```